tests/pmap_test.py

```diff
@@ -280,6 +280,106 @@ def testShardedDeviceTuple(self):
     w = jit(lambda x: list(x)[0])(y)
     self.assertAllClose(w, x, check_dtypes=False)
 
+  @jtu.skip_on_devices("cpu", "gpu")
+  def testCollectivePermute(self):
+    device_count = xla_bridge.device_count()
+    if device_count != 2:
```

Comment:
I might be tempted to generalize this test to an arbitrary number of devices. It should be well-defined for the n=1 case too if you are building a reversal or a rotation.

Reply:
Done.

Reply:
I sketched out an implementation, but will get it in a follow-up PR. Thanks for pointing this out!